mkosi

```diff
@@ -937,6 +937,10 @@ def reset_machine_id(args, workspace, run_build_script):
         machine_id = os.path.join(workspace, 'root', 'etc/machine-id')
         os.unlink(machine_id)
         open(machine_id, "w+b").close()
+        dbus_machine_id = os.path.join(workspace, 'root', 'var/lib/dbus/machine-id')
+        if os.path.exists(dbus_machine_id):
+            os.unlink(dbus_machine_id)
+            os.symlink('../../../etc/machine-id', dbus_machine_id)
```

Comment:
It seems reasonable to me to assume we are the only ones modifying the image, since we unshared the mount namespace and thus nobody else would be seeing this file. Moreover, I did not want to create the symlink if the file did not exist in the first place.

If you still prefer the unlink unconditionally approach, I can post a follow up PR, but I don't think it is necessary.